provider/thp_fragment_tool.py

```diff
@@ -0,0 +1,49 @@
+"""
+thp_fragment tool.
+
+This module is meant to copy, build and execute the thp_fragment tool.
+"""
+import os
+
+from avocado.utils import process
+
+from virttest import data_dir
+
+
+dst_dir = "/var/tmp"
+test_bin = "/var/tmp/thp_fragment"
+source_file = "thp_fragment.c"
+source_package = "thp_fragment.tar.gz"
```

Comment:
So the tarball only contains 1 file?

Reply:
Yes, the idea is to follow the same procedure as @zhenyzha implemented for the memory mapping tool in https://github.com/autotest/tp-qemu/pull/3772

Reply:
They are different, #3772 calls `utils_test.VMStress` to execute the stress tool, `VMStress` requires a tarball, but your PR executes the tool itself. For this one, compressing a single file and then decompressing it is redundant.

Reply:
@PaulYuuu so do you suggest to store the plain C script directly? That way we save the extraction step?

Reply:
I need to correct my previous comment, `VMStress` also supports a single file.

For this one, I think a tarball is unnecessary, compressing a single file and then decompressing it is redundant.

1. This is a single file and not a big file
2. The tarball is not encrypted, everyone can view the source code, but the C file is easier to view.
3. When the tool needs to be updated, then we have to create a new tarball
4. Reviewer/Users have to extract it to review the source code

So yes, I would suggest using the plain C file.

Reply:
(1/2) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.aarch64.page_64k.io-github-autotest-qemu.trans_hugepage.defrag.arm64-pci: STARTED
 (1/2) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.aarch64.page_64k.io-github-autotest-qemu.trans_hugepage.defrag.arm64-pci: FAIL: Failed on copying the tool package! (39.49 s)
 (2/2) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.aarch64.page_64k.io-github-autotest-qemu.trans_hugepage.relocated.arm64-pci: STARTED
 (2/2) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.aarch64.page_64k.io-github-autotest-qemu.trans_hugepage.relocated.arm64-pci: ERROR: [Errno 2] No such file or directory: '/home/kar/workspace/root/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-qemu/deps/thp_defrag_tool/thp_fragment.tar.gz' (57.85 s)
```
For 3987 you changed thp_fragment.tar.gz -> thp_fragment.c

Reply:
@zhenyzha I've still not updated the code because I am facing a permission error when using `shutil.copy2` let me go back to process.system till I obtain a response from @PaulYuuu

Reply:
@zhenyzha  @MiriamDeng please try again now, applying all the dependent patches, thanks !